src/commands/run.rs

```diff
@@ -0,0 +1,177 @@
+use super::CommandTrait;
+use crate::{
+    cli::{create_path_env, Config},
+    error::{MainError, PnError},
+    workspace,
+};
+use pipe_trait::Pipe;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs::File;
+use std::num::NonZeroI32;
+use std::path::Path;
+use std::{process, process::Stdio};
+
+/// Runs a defined package script.
+#[derive(Debug, clap::Args)]
+#[clap(rename_all = "kebab-case")]
+pub struct Run {
+    /// Name of the package script to run.
+    pub script: String, // Not OsString because it would be compared against package.json#scripts
+
+    /// Arguments to pass to the package script.
+    pub args: Vec<OsString>,
+}
+
+impl CommandTrait for Run {
```

Comment:
It is Rust's convention that a trait's name should match it's sole method name (for example: `Into::into`, `AsRef::as_ref`, `ToOwned::to_owned`, etc).

Therefore, I suggest the trait be renamed `ExecuteCommand` and the method `execute_command`.

Reply:
It turns out that this trait was a waste after all: https://github.com/pnpm/pn/pull/39/files#r1233114060

Reply:
Removed the trait entirely [here](https://github.com/pnpm/pn/pull/39/commits/576f80f65a8ce78bc6eaf2ab5cf3c4604e033d92)